core/include/detray/tools/navigator.hpp

```diff
@@ -428,31 +475,37 @@ class navigator {
      *
      * @param navigation [in, out] navigation state that contains the kernel
      */
+    DETRAY_HOST_DEVICE
     inline void set_next(state &navigation) const {
 
-        auto &kernel = navigation._kernel;
-
         // Sort distance to next & set navigation status
-        if (not kernel.candidates.empty()) {
+        if (not navigation.candidates().empty()) {
 
             // Take the nearest candidate first
-            std::sort(kernel.candidates.begin(), kernel.candidates.end());
-            kernel.next = kernel.candidates.begin();
+#if defined(__CUDACC__)
+            thrust::sort(thrust::seq, navigation.candidates().begin(),
+                         navigation.candidates().end());
+#else
+            std::sort(navigation.candidates().begin(),
+                      navigation.candidates().end());
+#endif
```

Comment:
You saw my comment earlier. I would prefer to use `thrust::sort(...)` in all cases here.

Reply:
Hmm let me explain what I understood about `thrust::sort`
```c++
thrust::sort(thrust::seq, iterator, iterator) // runs on single thread, called from device code
thrust::sort(iterator, iterator) // runs on multiple threads, called from host side
std::sort(iterator, iterator) // runs on single thread, called from host side
```
But I am not entirely sure that `thrust::sort(thrust::seq, iterator, iterator)` can only be called by device code. As I said earlier I could not include `thrust/sort.h` from host side